### PR TITLE
Solve struct ellipsoid caused the model to dark

### PR DESCRIPTION
**Pull request by verybigzhouhai**
Solved the problem that struct ellipsoid caused the model to darken on Qualcomm platform

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @verybigzhouhai!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/3dtile-dark/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by verybigzhouhai** [comment]
Fixed #7651

**Comment by verybigzhouhai** [comment]
@OmarShehata Hi，Now I am modifying the relevant code of the ellipsoid struct. Do you think it is acceptable?

**Comment by verybigzhouhai** [comment]
This is ready!

**Comment by OmarShehata** [comment]
Just to close the loop, this is the same fix as implemented in https://github.com/AnalyticalGraphicsInc/cesium/pull/7911 right?

**Comment by verybigzhouhai** [comment]
@OmarShehata yes, it's exactly the same.

**Comment by verybigzhouhai** [comment]
@OmarShehata here are two test failed, it doesn't seem to have anything to do with my code changes.Can you help me confirm it?

**Comment by verybigzhouhai** [comment]
@OmarShehata Will you review this?

**Comment by OmarShehata** [comment]
I can confirm this fixes the issue at https://github.com/AnalyticalGraphicsInc/cesium/issues/6622. I believe this also fixes https://github.com/AnalyticalGraphicsInc/cesium/issues/7651 and https://github.com/AnalyticalGraphicsInc/cesium/issues/7403.

Testing on the [physically based Sandcastle](https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/?src=Physically-Based%20Materials.html), in the hosted Sandcastle on Android:

65103522_484235775660795_8115887609747603456_n

In this branch:

65482229_472193313558138_6022643596195266560_n

@verybigzhouhai my main notes here are:

* I would avoid copying the value manually everywhere it's used in the code. Consider adding something in the shader constants that can be re-used like https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Shaders/Builtin/Constants/degreesPerRadian.glsl 
* I think this can be removed if it's not being used anywhere https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Shaders/Builtin/Structs/ellipsoid.glsl

**Comment by verybigzhouhai** [comment]
@OmarShehata I am very happy that the function is normal in your testing.
    I did use some constants to replace the structure at the beginning, but it caused a lot of unit tests to fail. I don't know how to avoid or need to modify a lot of unit test code, so I used the current modification method.
    Even if we use repeatable constants, I don't think this is the best way to modify them, because the structure itself is hoping to make the program simpler, so I think the best way to do this is to dynamically before compiling the shader. Replace the structure, but this approach may not be that simple, I have raised a similar issue here,  https://stackoverflow.com/questions/56479172/about-remove-replace-struct  , if we can do this, could make the code Keep the original elegance.
    What do you think about the above, if nothing, I will modify it again according to what you mentioned.

**Comment by OmarShehata** [comment]
@verybigzhouhai I'm curious about the unit tests that fail using that. Can you list a few of them and the errors you get?

**Comment by verybigzhouhai** [comment]
@OmarShehata I found some errors of my own, which will not lead to unit test failure now. Now I add two constants and delete files that do not need to be used. Updates have been submitted，you can check whether they meet the requirements.

File: Source/Shaders/Builtin/Constants/ellipsoidInverseRadii.glsl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * 0.1
```
Comment (OmarShehata) [inline Source/Shaders/Builtin/Constants/ellipsoidInverseRadii.glsl]: It looks like you copied this from the epsilon constants - I believe this should be a description of what this value is. For this, maybe it should say something like:

> The reciprocal of the radius of the WGS84 ellipsoid.

File: Source/Shaders/Builtin/Constants/ellipsoidRadii.glsl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/**
+ * 0.1
+ *
+ * @name czm_ellipsoidRadii
```
Comment (OmarShehata) [inline Source/Shaders/Builtin/Constants/ellipsoidRadii.glsl]: Same comment as in the inverse radii constant, maybe it should say something like:

> The radius of the WGS84 ellipsoid.

**Comment by OmarShehata** [comment]
@verybigzhouhai one more thing: Can you update `CHANGES.md` to add a bullet point under "fixes"? It should say something like:

> * Fixed a bug that caused 3D models to appear darker on Android devices. [#7944](https://github.com/AnalyticalGraphicsInc/cesium/pull/7944)

**Comment by verybigzhouhai** [comment]
Thank you very much for your guidance, @OmarShehata  now it should be ready to be merged.

**Comment by OmarShehata** [comment]
Great - I'm confident merging this since it's largely just rewriting structs to use builtin constants. It works in Chrome, Firefox, Edge, and IE11. It works correctly in Chrome on Android now. 

@lilleyse if you want to skim the changes later when you get a chance in case I missed any shader styling thing I wasn't aware of, that would be helpful!

**Comment by slchow** [comment]
If this is a first-time contribution, we'd like to share it. @verybigzhouhai do you have a Twitter handle?

**Comment by verybigzhouhai** [comment]
@slchow twitter handle @zhouhai19931228

**Pull request merged**